io-engine/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -910,17 +919,26 @@ impl<'n> Nexus<'n> {
         self.disconnect_device_from_channels(device_name.clone())
             .await?;
 
-        debug!("{self:?}: pausing...");
+        debug!("{self:?}: retire: pausing...");
         self.as_mut().pause().await?;
-        debug!("{self:?}: pausing ok");
+        debug!("{self:?}: retire: pausing ok");
 
         if let Some(child) = self.lookup_child_by_device(&device_name) {
+            // Cancel rebuild job for this child, if any.
+            if let Some(job) = child.rebuild_job() {
+                debug!("{self:?}: retire: stopping rebuild job...");
+                job.terminate().await.ok();
```

Comment:
looks like we're waiting for the job to complete here whilst still pausing, is this necessary?

Reply:
also what happens if rebuild succeeds (eg: let's say the last IO was sent before the replica got into a bad state)?
Conversely, what if this child was actually a rebuild source for another child? Probably that rebuild will fail, but can we somehow restart the rebuild from another child without getting rid of the log?